.github/workflows/functional-tests.yml

```diff
@@ -0,0 +1,41 @@
+name: Functional Tests
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+jobs:
+  test:
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 18
+
+    - name: Install root dependencies
+      run: npm install -g yarn && yarn install --immutable
+
+    - name: Build SDK
+      run: yarn build
+
+    - name: Install functional tests dependencies
+      run: cd tests/func-tests && yarn install --no-immutable
+
+    - name: Run tests
+      env:
+        NETWORK: sepolia
+        TEST_ALCHEMY_API_KEY: ${{ secrets.TEST_ALCHEMY_API_KEY }}
+        PUBLIC_API_URL: 'https://api.sandbox.x.immutable.com/v1'
+        TEST_STARK_CONTRACT_ADDRESS: '0x2d5C349fD8464DA06a3f90b4B0E9195F3d1b7F98'
+        TEST_REGISTRATION_CONTRACT_ADDRESS: '0xDbA6129C02E69405622fAdc3d5A7f8d23eac3b97'
+        TEST_TOKEN_ADDRESS: '0xfA5539fBEd27887EEbb2515672D80412D1A3ADa3'
+        TEST_PRIVATE_KEY1: ${{ secrets.TEST_PRIVATE_KEY1 }}
+        TEST_STARK_PRIVATE_KEY1: ${{ secrets.TEST_STARK_PRIVATE_KEY1 }}
+        TEST_PRIVATE_KEY2: ${{ secrets.TEST_PRIVATE_KEY2 }}
+        TEST_STARK_PRIVATE_KEY2: ${{ secrets.TEST_STARK_PRIVATE_KEY2 }}
```

Comment:
not blocking this PR from going in.
But I suggest follow up to rename `test_private_key1` to be something less ambiguous. e.g `test_wallet_private_key` ?

Reply:
Will use `TEST_WALLET1_PRIVATE_KEY` ... (and will change others to be consistent)

Reply:
Fixed.